### PR TITLE
Add exception to twitter oath rule for hashes in dependency files

### PR DESCRIPTION
**Pull request by 0xDC0DE**

### CODE REVIEW

**Comment by 0xDC0DE** [comment]
Fix for https://github.com/returntocorp/semgrep-rules/issues/2629

File: generic/secrets/security/detected-twitter-oauth.yaml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 rules:
 - id: detected-twitter-oauth
-  pattern-regex: |-
-    [tT][wW][iI][tT][tT][eE][rR].*['|"]?[0-9a-zA-Z]{35,44}['|"]?
+  patterns:
+    - pattern-regex: |-
+        [tT][wW][iI][tT][tT][eE][rR].*['|"]?[0-9a-zA-Z]{35,44}['|"]?
+    - pattern-not-regex: |-
```
Comment (LewisArdern) [inline generic/secrets/security/detected-twitter-oauth.yaml:6]: |- ?

**Review by LewisArdern — CHANGES_REQUESTED** [review]
👍 otherwise

**Review by LewisArdern — APPROVED** [review]
LGTM

**Pull request merged**